package.json

```diff
@@ -14,23 +14,58 @@
     "serve": "gatsby serve",
     "clean": "gatsby clean",
     "gh-pages": "gh-pages -d public",
-    "deploy": "npm run build && npm run gh-pages -- -m 'Updates: Manual build'"
+    "deploy": "npm run build && npm run gh-pages -- -m 'Updates: Manual build'",
+    "format": "prettier -w 'src/**/*.{ts,js}'",
+    "lint": "prettier --check 'src/**/*.{ts,js}'; eslint src --ext .ts,.js"
   },
   "dependencies": {
-    "@mantine/core": "4.2.11",
-    "@mantine/hooks": "4.2.11",
+    "@mantine/core": "4.2.12",
+    "@mantine/hooks": "4.2.12",
+    "@typescript-eslint/eslint-plugin": "5.30.7",
+    "@typescript-eslint/parser": "5.30.7",
+    "eslint": "8.20.0",
+    "eslint-config-prettier": "8.5.0",
     "gatsby": "4.17.2",
     "gatsby-plugin-google-analytics": "4.17.0",
     "gatsby-plugin-manifest": "4.17.0",
     "gatsby-plugin-mantine": "4.0.0",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "prettier": "2.7.1",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
     "react-helmet": "6.1.0",
     "tabler-icons-react": "1.52.0"
   },
   "devDependencies": {
     "eslint": "8.19.0",
     "eslint-plugin-react": "7.30.1",
     "gh-pages": "4.0.0"
+  },
+  "eslintConfig": {
```

Comment:
This should feature the config you had in `.eslintrc.yml` since it's "better" for this use case.

You want `env` to include `browser`, `es2021` (this one replaces `es6` - 2021 is the latest version so far, though there's not so many differences). You don't want `node` in `env`, since this is not a nodejs application.

Additionally you don't need any of the `overrides` since this repository doesn't contain any typescript. All `typescript` dependencies are useless here because of that.

The rest of the `.eslintrc.yml`, like the `extends: plugin:react/recommended` and plugins are also very useful here, since it's a React app.

You always need to tune your environment in JS to the app/lib purpose. In this case the purpose is totally different to the probot-template so it needs totally different settings.

Reply:
After looking through this, I think the config you had in `.eslintrc.yml` is fine for this application. You can either keep it in the `.eslintrc.yml` or move it into `package.json`, result will be the same. I think all you want to do here is to extend it with prettier in the `extends` config section + keep the prettier config.

Reply:
Isnt https://github.com/operate-first/operate-first.github.io/blob/54cb70f2ee3f19bac91450236fce124e34daa689/src/components/Layout.tsx a typescript component?

Reply:
yes, my bad.. they are, I didn't notice them. then you want TS for eslint :)